server.py

```
@@ -1,472 +1,472 @@
+                    follow_up_actions=follow_up_actions,
+                    metadata=llm_call.metadata,
+                )
+            finally:
+                storage.__exit__(None, None, None)
+    except AuthenticationError as e:
+        raise HTTPException(status_code=401, detail=e.message)
+    except litellm.exceptions.RateLimitError as e:
+        raise HTTPException(status_code=429, detail=e.message)
+    except Exception as e:
+        logging.error(f"Error in /api/chat: {e}", exc_info=True)
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)